samples/src/test/java/org/autorefactor/refactoring/rules/samples_in/AndroidViewHolderSample.java

```diff
@@ -0,0 +1,115 @@
+        return null;
+    }
+
+    @Override
+    public long getItemId(int position) {
+        return 0;
+    }
+
+    public static class Adapter1 extends AndroidViewHolderSample {
+        @Override
+        public View getView(int position, View convertView, ViewGroup parent) {
+            return null;
+        }
+    }
+
+    public static class Adapter2 extends AndroidViewHolderSample {
+        LayoutInflater mInflater;
+
+        @Override
+        public View getView(int position, View convertView, ViewGroup parent) {
+            convertView = mInflater.inflate(R.layout.your_layout, null);
+
+            TextView text = (TextView) convertView.findViewById(R.id.text);
+            text.setText("Position " + position);
+
+            return convertView;
+        }
+    }
+
+    public static class Adapter3 extends AndroidViewHolderSample {
+        LayoutInflater mInflater;
+
+        @Override
+        public View getView(int position, View convertView, ViewGroup parent) {
+            View v = mInflater.inflate(R.layout.your_layout, null);
+
+            TextView text = (TextView) v.findViewById(R.id.text);
+            text.setText("Position " + position);
+
+            return v;
+        }
+    }
+
+    public static class Adapter5 extends AndroidViewHolderSample {
+        LayoutInflater mInflater;
+
+        public View getView(int position, View convertView, ViewGroup parent) {
+            // Already using View Holder pattern
+            convertView = convertView == null ? mInflater.inflate(R.layout.your_layout, null) : convertView;
+
+            TextView text = (TextView) convertView.findViewById(R.id.text);
+            text.setText("Position " + position);
+
+            return convertView;
+        }
+    }
+
+    public static class Adapter7 extends AndroidViewHolderSample {
+        LayoutInflater inflater;
+
+        @Override
+        public View getView(final int position, final View convertView, final ViewGroup parent) {
+            View rootView = convertView;
+            final int itemViewType = getItemViewType(position);
+            switch (itemViewType) {
+            case 0:
+                if (rootView != null)
+                    return rootView;
+                rootView = inflater.inflate(android.R.layout.simple_list_item_1, parent, false);
+                break;
+            }
+            return rootView;
+        }
+    }
+
+    /* TODO low priority VieHolder cornercase
+    public static class CornerCase extends ViewHolderSample {
+        LayoutInflater inflater;
+
+        @Override
+        public View getView(final int position, final View convertView, final ViewGroup parent) {
+            View rootView = convertView;
+            if (rootView != null)
+                return rootView;
+            rootView = inflater.inflate(android.R.layout.simple_list_item_1, parent, false);
+            return rootView;
+        }
+    }*/
+
+    private static class R {
```

Comment:
Yes! GUI resources are defined using XML and can have the attribute id. This id is later used to get that resource in the code. ``R.id.<id_of_resource>`` is the way  it's done.
In the test it is not really necessary since it is nothing more than an ``int``. However, it makes it look like Android code.